specification/operationalinsights/data-plane/Microsoft.OperationalInsights/v1/OperationalInsights.json

```diff
@@ -0,0 +1,345 @@
+                "description": "Executes an Analytics query for data. [Here](/documentation/2-Using-the-API/Query) is an example for using POST with an Analytics query.",
+                "x-ms-examples": {
+                    "simple-query": {
+                        "$ref": "examples/oms-post-example.json"
+                    },
+                    "cross-workspace": {
+                        "$ref": "examples/oms-post-cross-workspace-example.json"
+                    }
+                },
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/workspace-id"
+                    },
+                    {
+                        "$ref": "#/parameters/query-body"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK. The API call succeeded and the Analytics query result is in the response payload",
+                        "schema": {
+                            "$ref": "#/definitions/query-results"
+                        }
+                    },
+                    "default": {
+                        "description": "An error response object.",
+                        "schema": {
+                            "$ref": "#/definitions/error-response"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "parameters": {
+        "workspace-id": {
+            "name": "workspace-id",
+            "description": "ID of the workspace. This is Workspace ID from the Properties blade in the Azure portal.",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-parameter-location": "client"
+        },
+        "query-param": {
+            "name": "query",
+            "in": "query",
+            "required": true,
+            "x-ms-parameter-location": "method",
+            "description": "The Analytics query. Learn more about the [Analytics query syntax](https://azure.microsoft.com/documentation/articles/app-insights-analytics-reference/)",
+            "type": "string"
+        },
+        "query-body": {
+            "name": "body",
+            "in": "body",
+            "description": "The Analytics query. Learn more about the [Analytics query syntax](https://azure.microsoft.com/documentation/articles/app-insights-analytics-reference/)",
+            "required": true,
+            "schema": {
+                "$ref": "#/definitions/query-body"
+            },
+            "x-ms-parameter-location": "method"
+        },
+        "timespan-param": {
+            "name": "timespan",
+            "in": "query",
+            "required": false,
+            "x-ms-parameter-location": "method",
+            "description": "Optional. The timespan over which to query data. This is an ISO8601 time period value.  This timespan is applied in addition to any that are specified in the query expression.",
```

Comment:
Oooh. also, add a `"format": "duration",` and we will generate a Duration or Timespan instead of a string.